include/na/azac/AStarPlacer.hpp

```diff
@@ -0,0 +1,647 @@
+   * the standard deviation of the differences between the current and target
+   * sites of the atoms to be moved in every orientation.
+   * @details This factor is multiplied with the sum of standard deviations to
+   * adjust the influence of this term. Setting it to 0.0 disables this term
+   * and, if the lookahead is also disabled, resulting in an admissible
+   * heuristic. However, this leads to a vast exploration of the search tree and
+   * usually results in a huge number of nodes visited.
+   */
+  float deepeningFactor_ = 0.8F;
+  /**
+   * @brief Before the sum of standard deviations is multiplied with the number
+   * of unplaced nodes and @ref deepeningFactor_, this value is added to the sum
+   * to amplify the influence of the unplaced nodes count.
+   * @see deepeningFactor_
+   */
+  float deepeningValue_ = 0.2F;
+  /**
+   * @brief The cost function can consider the distance of atoms to their
+   * interaction partner in the next layer.
+   * @details This factor is multiplied with the distance to adjust the
+   * influence of this term. Setting it to 0.0 disables the lookahead entirely.
+   * A factor of 1.0 implies that the lookahead is as important as the distance
+   * to the target site, which is usually not desired.
+   */
+  float lookaheadFactor_ = 0.2F;
+  /**
+   * @brief The reuse level corresponds to the estimated extra fidelity loss due
+   * to the extra trap transfers when the atom is not reused and instead moved
+   * to the storage zone and back to the entanglement zone.
+   * @details It is subtracted from the cost for the reuse option to favor this
+   * option over the non-reuse options.
+   */
+  float reuseLevel_ = 5.0F;
+  /**
+   * @brief The maximum number of nodes that are allowed to be visited in the A*
+   * search tree.
+   * @detsils If this number is exceeded, the search is aborted and an error is
+   * raised. In the current implementation, one node roughly consumes 120 Byte.
+   * Hence, allowing 50,000,000 nodes results in memory consumption of about 6
+   * GB plus the size of the rest of the data structures.
+   */
+  size_t maxNodes_ = 50'000'000;
+  /**
+   * @brief When placing atoms after a rydberg layer back in the storage zone,
+   * this struct stores for every such atom all required information, i.e., the
+   * current site and potential target sites ordered by distance (ascending).
+   */
+  struct AtomJob {
+    /// The atom to be placed
+    qc::Qubit atom;
+    /// The current site of the atom
+    DiscreteSite currentSite;
+    /// The minimum lookahead distance
+    float meanLookaheadCost = 0.0F;
+    /// A struct describing one potential target site
+    struct Option {
+      /// The target site
+      DiscreteSite site;
+      /**
+       * When this flag is set to false, it indicates that the atom should not
+       * move at all and remain in the entanglement zone. Then the attribute
+       * site is ignored.
+       */
```

Comment:
Maybe I am not reading this right or have a knot in my mind, but isn't this mixed up?
`reuse=true` should imply that the atom must not move, right?